packages/polymath-ui/src/helpers.js

```diff
@@ -49,6 +49,19 @@ export const thousandsDelimiter = (v: number) => {
   );
 };
 
+export const formatLargeNumber = (
+  data: number,
+  precision: number,
+  decimals: number
+) => {
+  const str = data.toString();
+  if (str.indexOf('e') !== -1) {
+    return Number(data.toPrecision(precision)).toExponential(decimals);
+  } else {
+    return parseFloat(data.toFixed(decimals));
+  }
+};
+
```

Comment:
@RafaelVidaurre if you could take a look at that as you worked with the numbers. Shouldn't BigNumber lib that we are using handle this?
Also if this function is actually necessary we should have it in `shared` and `new-shared` instead of here, that file is deprecated (might be worth adding a comment in there).

Reply:
I've made these changes